projects/jupyter-server-ydoc/jupyter_server_ydoc/handlers.py

```diff
@@ -81,6 +82,21 @@ async def prepare(self):
             if self._websocket_server.room_exists(self._room_id):
                 self.room: YRoom = await self._websocket_server.get_room(self._room_id)
             else:
+                # Logging exceptions, instead of raising them here to ensure
+                # that the y-rooms stay alive even after an exception is seen.
+                def exception_logger(exception: Exception, log: Logger) -> bool:
+                    """A function that catches any exceptions raised in the websocket
+                    server and logs them.
+
+                    The protects the y-room's task group from cancelling
+                    anytime an exception is raised.
+                    """
+                    log.error(
+                        f"Document Room Exception, (room_id={self._room_id or 'unknown'}): ",
+                        exc_info=exception,
+                    )
+                    return True
```

Comment:
This is almost the same function as in `websocketserver.py`, except for the logging message. I think this should be factored out, with an extra-argument for the message.

Reply:
For sure. Since this is an internal definition right now, let's factor this out in a later PR. I want to proceed with merging, releasing, and testing.